Src/VimCore/CommonOperations.fs

```diff
@@ -247,10 +247,23 @@ type internal CommonOperations
 
     // Convert any virtual spaces to real normalized spaces
     member x.FillInVirtualSpace () =
+
+        // This is an awkward situation if 'expandtab' is in effect because
+        // Visual Studio uses tabs when filling in virtual space, but vim
+        // doesn't. VsVim needs to insert tabs when filling in leading virtual
+        // space to emulate the way vim copies leading tabs from surrounding
+        // lines, but shouldn't insert tabs when filling in non-leading virtual
+        // space because vim never inserts tabs in that situation.
         if x.CaretVirtualPoint.IsInVirtualSpace then
             let blanks: string = 
                 let blanks = StringUtil.RepeatChar x.CaretVirtualColumn.VirtualSpaces ' '
-                x.NormalizeBlanks blanks
+                if x.CaretPoint = x.CaretLine.Start then
+
+                    // The line is completely empty so use tabs if appropriate.
+                    let spacesToColumn = x.GetSpacesToPoint x.CaretPoint

```

Comment:
Is there any case where columns vs. points would return a different answer here? Given we're in virtual space pretty sure this is no because there can't be any wide code points.

Reply:
This computation would work correctly in the presence of wide code points, but given the previous conditions, the value of `spacesToColumn` will always be zero so the calculation is a formality.